File: LeetCode Problems/src/leetcode/ArraysStrings/TwoSum.java
package leetcode.ArraysStrings;

import java.util.Arrays;

public class TwoSum {

	public static void main(String[] args) {

		int[] numbers = { 2, 4, 5, 7, 8, 90 };
		int target = -100;

		System.out.println(Arrays.toString(findNums(numbers, target)));
	}

	private static int[] findNums(int[] numbers, int target) {

		int start = 0, end = numbers.length - 1;
		while (start < end) {
//			System.out.println(start + " " + end);
			if (target == numbers[start] + numbers[end]) {
				System.out.println(start + " result " + end);
				return new int[] { start, end };
			} else if (target > numbers[start] + numbers[end]) {
				start++;
			} else {
				end--;
			}
		}
		return new int[] { 0 };
	}

}

File: LeetCode Problems/src/Easy/TwoSum.py
def twoSum(nums, target):
        hash = {}
        for item in nums:
            hash[item] = hash.get(item, 0) + 1

        for item in nums:
            if (target - item) in hash:
                return (target - item, item)

        return (0, 0)

nums = [2, 7, 11, 15]
target = 9
result = twoSum(nums, target)
print(result)